crates/amf0/src/value.rs

```diff
@@ -1,18 +1,19 @@
 //! AMF0 value types.
 
-use std::borrow::Cow;
-use std::collections::HashMap;
+use std::collections::BTreeMap;
 use std::io;
+use std::marker::PhantomData;
 
 use scuffle_bytes_util::StringCow;
+use serde::de::IntoDeserializer;
 
 use crate::Amf0Error;
 use crate::encoder::Amf0Encoder;
 
 /// Represents any AMF0 object.
-pub type Amf0Object<'a> = HashMap<StringCow<'a>, Amf0Value<'a>>;
+pub type Amf0Object<'a> = BTreeMap<StringCow<'a>, Amf0Value<'a>>;
```

Comment:
We should make this a wrapper instead of a type def.

So right now this is assigned to be a BTreeMap, this is less than ideal because in the future we might want to preserve order strictly (ie an `IndexMap`) and that would be a breaking change. Instead we should make a new structure called `Amf0Object` and then internally we have a `BTreeMap` and then we provide a bunch of helper functions reexporting the BTreeMap's behaviours (`len`, `is_empty`, `iter`, `into_iter`, `get`, `get_mut`, `remove`, `insert`, `from_iter`, etc)